provider/priority_data.go

```diff
@@ -0,0 +1,55 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/firehydrant/terraform-provider-firehydrant/firehydrant"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourcePriority() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataFireHydrantPriority,
+		Schema: map[string]*schema.Schema{
+			"slug": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"default": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
```

Comment:
Minor nit: We try to keep these in alphabetical order within a given required/optional/computed section. 
```suggestion
			// Required
			"slug": {
				Type:     schema.TypeString,
				Required: true,
			},
			
			// Computed
			"default": {
				Type:     schema.TypeBool,
				Computed: true,
			},
			"description": {
				Type:     schema.TypeString,
				Computed: true,
			},
```